### PR TITLE
Produce errors more consistently in operator policy package

### LINKED ISSUE #662: Errors: internal/policy/operator
As part of cleaning up how errors are wrapped and handled, this package needs attention.

### PR DESCRIPTION
**Pull request by bcrochet**
* Remove errors.go
* Make sure error types are not leaked from the checks
* Remove extraneous logging

Fixes #662

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

File: certification/internal/policy/operator/scorecard_olm_suite_test.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ var _ = Describe("ScorecardBasicCheck", func() {
 			})
 			It("Should not pass Validate", func() {
 				ok, err := scorecardOlmSuiteCheck.Validate(context.TODO(), certification.ImageReference{ImageURI: "dummy/image"})
-				Expect(err).ToNot(HaveOccurred())
+				Expect(err).To(HaveOccurred())
```
Comment (acornett21) [inline certification/internal/policy/operator/scorecard_olm_suite_test.go:91]: Same question as the other scorecard test.

Comment (bcrochet) [reply certification/internal/policy/operator/scorecard_olm_suite_test.go:91]: It's effectively the same.

File: certification/internal/policy/operator/scorecard_basic_check_spec_test.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ var _ = Describe("ScorecardBasicCheck", func() {
 			})
 			It("Should not pass Validate", func() {
 				ok, err := scorecardBasicCheck.Validate(context.TODO(), certification.ImageReference{ImageURI: "dummy/image"})
-				Expect(err).ToNot(HaveOccurred())
+				Expect(err).To(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (bcrochet) [reply certification/internal/policy/operator/scorecard_basic_check_spec_test.go:91]: Well, it wasn't checking the error for Validate. It probably should have been.

[the remark this answers is not in the document]
Comment (bcrochet) [reply certification/internal/policy/operator/scorecard_basic_check_spec_test.go:91]: I guess something bad changed. Getting a panic in check operator. Putting this on hold while I investigate.

**Review by komish — APPROVED** [review]
/lgtm

**Pull request merged**